mailbox/event/event-memory/src/test/java/org/apache/james/mailbox/events/delivery/InVmEventDeliveryTest.java

```diff
@@ -53,6 +57,77 @@ void setUp() {
         inVmEventDelivery = new InVmEventDelivery(new NoopMetricFactory());
     }
 
+    @Nested
+    class ErrorHandling {
+
+        @Test
+        void deliverShouldReturnSuccessResultWhenSynchronousListenerFailsLEMaxRetry() {
+            when(listener.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.SYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener).event(event);
+
+            assertThatCode(() -> inVmEventDelivery.deliver(ImmutableList.of(listener), event)
+                    .allListenerFuture()
+                    .block())
+                .doesNotThrowAnyException();
+        }
+
+
+        @Test
+        void deliverShouldReturnErrorResultWhenSynchronousListenerFailsGTMaxRetry() {
+            when(listener.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.SYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener).event(event);
+
+            assertThatThrownBy(() -> inVmEventDelivery.deliver(ImmutableList.of(listener), event)
+                    .allListenerFuture()
+                    .block())
+                .isInstanceOf(RuntimeException.class);
```

Comment:
IMO you are note supposed to throw here (see later EventDeadLetter where we would be storing event + group)

Reply:
if my `eventDelivery.deliver()` always returns a `success Mono`, how could I know when I should store my event in deadletter? because currently It always returns `success Mono` even failed to delivers to listeners...

Reply:
Right we could use `doOnError` then switch it to a success mono in the event bus.